src/actions/connectionsActions.js

```diff
@@ -106,23 +98,25 @@ export const updateConnectionsAction = (theWalletId?: ?string = null) => {
       }
     });
 
-    const updatedContacts = uniqBy(contacts.concat(allContacts), 'id')
+    const updatedContacts = uniqBy(contacts, 'id')
       .filter(({ id }) => !removeContacts.includes(id));
-    const updatedInvitations = uniqBy(invitations.concat(allInvitations), 'id')
+    const updatedInvitations = uniqBy(invitations, 'id')
       .filter(({ id }) => !removeInvitations.includes(id));
 
-    dispatch({
+    dispatch(saveDbAction('invitations', { invitations: updatedInvitations }, true));
+    dispatch(saveDbAction('contacts', { contacts: updatedContacts }, true));
+
+    await dispatch({
       type: UPDATE_INVITATIONS,
       payload: updatedInvitations,
     });
-    dispatch(saveDbAction('invitations', { invitations: updatedInvitations }, true));
 
-    dispatch({
+    await dispatch({
```

Comment:
what the point of using awaits here?

Reply:
would it matter if they are async but in order ?   I thought this  await dispatch(getExistingChatsAction()); was needed after state is set

Reply:
As I remember there is no need to use `await` if you simply dispatch the object.
But you can easily check this by dispatching some change and checking if it was changed immediately after the call:
```javascript
dispatch({ type: UPDATE_SOMETHING });
const { wasUpdated } = getState().someReducer;
if (!wasUpdate) console.log('I was right!');
```